client/my-sites/hosting/web-server-settings-card/index.tsx

```diff
@@ -49,57 +50,73 @@ const InputPlaceholder = styled( LoadingPlaceholder )( {
 	marginBottom: '1em',
 } );
 
-const WebServerSettingsCard = ( {
-	disabled,
-	isGettingGeoAffinity,
-	isGettingPhpVersion,
-	isGettingStaticFile404,
-	isGettingWpVersion,
-	isUpdatingPhpVersion,
-	isUpdatingStaticFile404,
-	isUpdatingWpVersion,
-	isWpcomStagingSite,
-	siteId,
-	selectedSiteSlug,
-	geoAffinity,
-	staticFile404,
-	translate,
-	updatePhpVersion,
-	updateStaticFile404,
-	updateWpVersion,
-	phpVersion,
-	wpVersion,
-} ) => {
+type WebServerSettingsCardProps = {
+	disabled?: boolean;
+};
+
+export default function WebServerSettingsCard( { disabled }: WebServerSettingsCardProps ) {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const siteId = useSelector( getSelectedSiteId );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+
+	const isWpcomStagingSite = useSelector( ( state ) => isSiteWpcomStaging( state, siteId ) );
+	const geoAffinity = useSelector( ( state ) => getAtomicHostingGeoAffinity( state, siteId ) );
+	const phpVersion = useSelector( ( state ) => getAtomicHostingPhpVersion( state, siteId ) );
+	const wpVersion = useSelector( ( state ) => getAtomicHostingWpVersion( state, siteId ) );
+	const staticFile404 = useSelector( ( state ) => getAtomicHostingStaticFile404( state, siteId ) );
+
+	const isGettingGeoAffinity = useSelector( ( state ) =>
+		isFetchingAtomicHostingGeoAffinity( state, siteId )
+	);
+	const isGettingWpVersion = useSelector( ( state ) =>
+		isFetchingAtomicHostingWpVersion( state, siteId )
+	);
+	const isGettingStaticFile404 = ! disabled && ! staticFile404;
+	const isUpdatingPhpVersion = useSelector(
+		( state ) => getRequest( state, updateAtomicPhpVersion( siteId, null ) )?.isLoading ?? false
+	);
+	const isUpdatingStaticFile404 = useSelector(
+		( state ) => getRequest( state, updateAtomicStaticFile404( siteId, null ) )?.isLoading ?? false
+	);
+	const isUpdatingWpVersion = useSelector(
+		( state ) => getRequest( state, updateAtomicWpVersion( siteId, null ) )?.isLoading ?? false
+	);
```

Comment:
This is beyond the scope of this PR, but I can't help notice how much more compact this code would be if we used `react-query` to fetch this data instead.

Reply:
I created https://github.com/Automattic/dotcom-forge/issues/8924 for this